qa/tasks/ceph_test_case.py

```diff
@@ -13,7 +14,32 @@
 class TestTimeoutError(RuntimeError):
     pass
 
-class CephTestCase(unittest.TestCase):
+class RunCephCmd:
+
+    def run_ceph_cmd(self, *args, **kwargs):
+        if kwargs.get('args') is None and args:
+            if len(args) == 1:
+                args = args[0]
+            kwargs['args'] = args
+        return self.mon_manager.run_cluster_cmd(**kwargs)
+
+    def get_ceph_cmd_result(self, *args, **kwargs):
+        if kwargs.get('args') is None and args:
+            if len(args) == 1:
+                args = args[0]
+            kwargs['args'] = args
+        return self.run_ceph_cmd(**kwargs).exitstatus
+
+    def get_ceph_cmd_stdout(self, *args, **kwargs):
+        if kwargs.get('args') is None and args:
+            if len(args) == 1:
+                args = args[0]
+            kwargs['args'] = args
+        kwargs['stdout'] = kwargs.pop('stdout', StringIO())
+        return self.run_ceph_cmd(**kwargs).stdout.getvalue()
+
+
+class CephTestCase(unittest.TestCase, RunCephCmd):
```

Comment:
this didn't seem to work either
```
2023-08-19T19:25:58.769 INFO:tasks.cephfs_test_runner:test_cluster_info (tasks.cephfs.test_nfs.TestNFS) ... ERROR
2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:
2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:======================================================================
2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:ERROR: test_cluster_info (tasks.cephfs.test_nfs.TestNFS)
2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 578, in test_cluster_info
2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:    self._test_create_cluster()
2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 156, in _test_create_cluster
2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:    cluster_create = self._nfs_complete_cmd(
2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 25, in _nfs_complete_cmd
2023-08-19T19:25:58.772 INFO:tasks.cephfs_test_runner:    return self.run_ceph_cmd(args=f"nfs {cmd}", stdout=StringIO(),
2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/ceph_test_case.py", line 24, in run_ceph_cmd
2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:    return self.mon_manager.run_cluster_cmd(**kwargs)
2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:AttributeError: 'NoneType' object has no attribute 'run_cluster_cmd'
2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:
2023-08-19T19:25:58.774 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2023-08-19T19:25:58.774 INFO:tasks.cephfs_test_runner:Ran 1 test in 42.921s
```
https://pulpito.ceph.com/adking-2023-08-19_17:52:39-orch:cephadm-wip-adk-testing-2023-08-19-1107-distro-default-smithi/7373833

Reply:
> Do we want to consider reverting the test_nfs test to the old way of running cluster commands or do you have another idea @rishabh-d-dave

I think this can be worked out. Give me some time, I'll spend it with how test_nfs.py runs with teuthology.

test_nfs.py can't be run with vstart_runner.py, right?

Reply:
yeah, nfs test cases are incompatible with vstart_runner

Reply:
> I think this can be worked out. Give me some time, I'll spend it with how test_nfs.py runs with teuthology.

Do you have an update on the fix @rishabh-d-dave.

Reply:
> this didn't seem to work either
> ```
> 2023-08-19T19:25:58.769 INFO:tasks.cephfs_test_runner:test_cluster_info (tasks.cephfs.test_nfs.TestNFS) ... ERROR
> 2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:
> 2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:======================================================================
> 2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:ERROR: test_cluster_info (tasks.cephfs.test_nfs.TestNFS)
> 2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
> 2023-08-19T19:25:58.770 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
> 2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 578, in test_cluster_info
> 2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:    self._test_create_cluster()
> 2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 156, in _test_create_cluster
> 2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:    cluster_create = self._nfs_complete_cmd(
> 2023-08-19T19:25:58.771 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/cephfs/test_nfs.py", line 25, in _nfs_complete_cmd
> 2023-08-19T19:25:58.772 INFO:tasks.cephfs_test_runner:    return self.run_ceph_cmd(args=f"nfs {cmd}", stdout=StringIO(),
> 2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_db1309a82b4354eab4e556918ba42cd57571830b/qa/tasks/ceph_test_case.py", line 24, in run_ceph_cmd
> 2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:    return self.mon_manager.run_cluster_cmd(**kwargs)
> 2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:AttributeError: 'NoneType' object has no attribute 'run_cluster_cmd'
> 2023-08-19T19:25:58.773 INFO:tasks.cephfs_test_runner:
> 2023-08-19T19:25:58.774 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
> 2023-08-19T19:25:58.774 INFO:tasks.cephfs_test_runner:Ran 1 test in 42.921s
> ```
> https://pulpito.ceph.com/adking-2023-08-19_17:52:39-orch:cephadm-wip-adk-testing-2023-08-19-1107-distro-default-smithi/7373833

@rishabh-d-dave also bring in _init_mon_manager() perhaps?